### DISCUSSION TITLE
Fixed station exception capture

### DISCUSSION THREAD

**Discussion by noob3-3** [semuconsulting/pygnssutils#81] [Q&A]
How can GNSSSocketServer capture exceptions, such as:
```
July 31 13:59:17 n1001-Default-string python3[12089]:   File "/usr/lib/python3.10/socketserver.py", line 747, in __init__
July 31 13:59:17 n1001-Default-string python3[12089]:     self.handle()
July 31 13:59:17 n1001-Default-string python3[12089]:   File "/home/n100_1/.local/lib/python3.10/site-packages/pygnssutils/socket_server.py", line 285, in handle
July 31 13:59:17 n1001-Default-string python3[12089]:     self._write_from_mq()
July 31 13:59:17 n1001-Default-string python3[12089]:   File "/home/n100_1/.local/lib/python3.10/site-packages/pygnssutils/socket_server.py", line 409, in _write_from_mq
July 31 13:59:17 n1001-Default-string python3[12089]:     self.wfile.write(raw)
July 31 13:59:17 n1001-Default-string python3[12089]:   File "/usr/lib/python3.10/socketserver.py", line 826, in write
July 31 13:59:17 n1001-Default-string python3[12089]:     self._sock.sendall(b)
July 31 13:59:17 n1001-Default-string python3[12089]: OSError: [Errno 113] No route to host
July 31 13:59:17 n1001-Default-string python3[12089]: ----------------------------------------
```
Or when a USB serial port gets disconnected, catch the exception and automatically restart the service.
My code
```
server = pygnssutils.gnssserver.GNSSSocketServer(inport="com3",baudrate="115200",ntripmode=2)
server.run()
```

**Comment by semuadmin** [comment]
Hi @noob3-3,

There's currently no automatic restart facility in the pygnssutils GNSSSocketServer utility, and I think you'd to consider the circumstances in which an automated restart would be appropriate. For example, I'm not sure I'd want an automated restart in the event that a USB serial port got physically disconnected (_as that would presumably require some manual intervention to resolve?_); but for temporary network outages, sure.

I can look into this in a future release.

**Reply by noob3-3** [reply]
I want to catch the exception myself and restart it, but after run(), I can't seem to catch the exception in the child thread

**Comment by noob3-3** [comment]
@semuadmin 1.1.1versions ，run `gnssserver --inport "/dev/tty.usbmodem101" --baudrate 115200 --hostip 192.168.0.27 --outport 50012 --verbosity 2`
show err

`TypeError: GNSSStreamer.__init__() missing 2 required positional arguments: 'app' and 'stream'`

**Reply by semuadmin** [reply]
Raised as Issue #90, will be fixed in #91
